source/robot_model/include/robot_model/QPSolver.hpp

```diff
@@ -0,0 +1,93 @@
+
+#include <OsqpEigen/OsqpEigen.h>
+#include <pinocchio/multibody/data.hpp>
+#include <state_representation/space/cartesian/CartesianPose.hpp>
+#include <state_representation/space/joint/JointPositions.hpp>
+
+using namespace std::chrono_literals;
+
+namespace robot_model {
+
+/**
+ * @brief parameters for the inverse velocity kinematics function
+ * @param alpha gain associated to the time slack variable
+ * @param proportional_gain gain to weight the cartesian coordinates in the gradient
+ * @param linear_velocity_limit maximum linear velocity allowed in Cartesian space (m/s)
+ * @param angular_velocity_limit maximum angular velocity allowed in Cartesian space (rad/s)
+ * @param period of the control loop (ns)
+ */
+struct QPInverseVelocityParameters {
+  double alpha = 0.1;
+  double proportional_gain = 1.0;
+  double linear_velocity_limit = 2.0;
+  double angular_velocity_limit = 2.0;
+  std::chrono::nanoseconds dt = 1000ns;
+};
+
+/**
+ * @class QPSolver
+ * @brief A class to solve the Quadratic Programming problem (QP) for the robot model
+*/
+class QPSolver {
+private:
+  // @format:off
+  unsigned nb_joints_;                                                         ///< number of joints in the robot model
+  pinocchio::ModelTpl<double>::ConfigVectorType lower_position_limit_;         ///< lower position limits of the joints from the URDF
```

Comment:
Is this the same as `Eigen::MatrixXd`? If yes, it would be more readable to use eigen

Reply:
Initially I set it as `Eigen::MatrixXd` but the compiler complained, I then verified that it is not of type  `Eigen::MatrixXd` but rather a custom type in `pinocchio`.